tools/wptserve/wptserve/pipes.py

```diff
@@ -471,21 +471,24 @@ def config_replacement(match):
             raise Exception("Undefined template variable %s" % field)
 
         while tokens:
-            ttype, field = tokens.popleft()
+            ttype, ffield = tokens.popleft()
             if ttype == "index":
-                value = value[field]
+                value = value[ffield]
             elif ttype == "arguments":
-                value = value(request, *field)
+                value = value(request, *ffield)
             else:
                 raise Exception(
-                    "unexpected token type %s (token '%r'), expected ident or arguments" % (ttype, field)
+                    "unexpected token type %s (token '%r'), expected ident or arguments" % (ttype, ffield)
                 )
 
         assert isinstance(value, (int, (binary_type, text_type))), tokens
 
         if variable is not None:
             variables[variable] = value
 
+        if field == "GET" and not isinstance(value, str):
```

Comment:
I'm not sure about the ramifications this might have. Would any case where this code path is hit previous be totally broken, or is it some more subtle change of behavior? If subtle, then we should look at differences in test results after this lands.

Also, I guess we'll need to import this change into Chromium ASAP after it lands here, or the tests won't work? (Tests and tools are imported separately, which is a bit of a problem in cases like this.)

Reply:
Alright, sounds like we can try landing this without running all the tests against the change first.

Reply:
This doesn't make sense to me at all, and I can't reproduce it either. There's already
```py
if isinstance(value, binary_type):
    value = value.decode("utf-8")
```

below, which I can only imagine makes your change pointless.

Can you explain what exactly is going wrong?

Reply:
And is there any reasons that we don't have any tests added for whatever the failure was?

Reply:
@andypaicu Oh, I thought we had unit tests for this, but no, only the replacement tokenizer.